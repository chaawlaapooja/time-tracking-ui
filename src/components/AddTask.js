import React,{useState} from 'react';
import CustomModal from './Modal';

const AddTask = () => {
    const [isOpen, setIsOpen] = useState(false)
    const close_modal = () => {
        setIsOpen(false)
    }
    return(
        <div>
            <button className="button" onClick={()=>setIsOpen(true)}>+ Add Task</button>
            {isOpen?<CustomModal close_modal={close_modal} purpose='add' name='' tags={[]} />:undefined}
        </div>
    )
}

export default AddTask