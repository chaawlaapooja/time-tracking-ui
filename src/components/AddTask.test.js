import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import AddTask from './AddTask';

configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';


describe('Add Task Component testing', ()=>{
    let wrapper;
    
    beforeEach(()=>{
        wrapper = render(<AddTask/>)
    })

    test('render the add task button',()=>{
        expect(wrapper.find('button').text()).toBe('+ Add Task')
    })
    test('change state to open modal on button click', ()=>{
        wrapper = shallow(<AddTask/>)
        wrapper.find('.button').simulate('click')
        expect(wrapper.find('#custommodal')).toHaveLength(0)
    })
    it('matches snapshot',()=>{
        const tree = renderer.create(<AddTask />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})