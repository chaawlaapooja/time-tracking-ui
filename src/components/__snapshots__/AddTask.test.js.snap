// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Task Component testing matches snapshot 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
  >
    + Add Task
  </button>
</div>
`;
